pkg/operator/ceph/object/objectstore.go

```diff
@@ -235,7 +241,25 @@ func deletePools(context *Context, lastStore bool) error {
 	return nil
 }
 
-func createPools(context *Context, metadataSpec, dataSpec model.Pool) error {
+func createPools(context *Context, spec cephv1.ObjectStoreSpec, metadataSpec, dataSpec model.Pool) error {
+	if emptyPool(spec.DataPool) && emptyPool(spec.MetadataPool) {
+		logger.Infof("no pools specified for the object store, checking for their existence...")
+		pools := append(metadataPools, dataPools...)
+		pools = append(pools, rootPool)
+		var missingPools []string
+		for _, pool := range pools {
+			poolName := poolName(context.Name, pool)
+			_, err := ceph.GetPoolDetails(context.Context, context.ClusterName, poolName)
+			if err != nil {
+				logger.Debugf("failed to find pool %q. %v", poolName, err)
+				missingPools = append(missingPools, poolName)
+			}
+		}
+		if len(missingPools) > 0 {
+			return fmt.Errorf("object store pools are missing: %v", missingPools)
```

Comment:
```suggestion
			return errors.Wrapf(err, "object store pools are missing: %v", missingPools)
```

Reply:
there is no err to wrap here

Reply:
Oops sorry, then `errors.Errorf` should be used.

Reply:
How about we merge this for now? There are 500+ instances of fmt.Errorf() and only 100+ of errors.Errorf so we can do a bulk change separately.

Reply:
Fine.